### PR TITLE
Add ts_vector column for better searching

### PR DESCRIPTION
**Pull request by marija04**

### CODE REVIEW

File: supabase/migrations/20241127203353_searchable_index_col.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
+ALTER TABLE topics
+    ADD COLUMN searchable_index_col tsvector;
+
+CREATE INDEX search_idx ON topics USING gin(searchable_index_col);
```
Comment (erik-beus) [inline supabase/migrations/20241127203353_searchable_index_col.sql:4]: Is `gin` the preferred index type for tsvector columns?

File: supabase/migrations/20241127203353_searchable_index_col.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    EXECUTE FUNCTION private.update_searchable_index_col();
+
+-- UPDATE
+--     topics t
+-- SET
+--     searchable_index_col = to_tsvector('english', t.name || ' ' || m.content)
+-- FROM
+--     messages m
+-- WHERE
+--     t.first_message_id = m.id;
```
Comment (erik-beus) [inline supabase/migrations/20241127203353_searchable_index_col.sql]:
```suggestion
```

File: supabase/migrations/20241127203353_searchable_index_col.sql [hunk 3]
```diff
@@ -0,0 +1,36 @@
+ALTER TABLE topics
+    ADD COLUMN searchable_index_col tsvector;
+
+CREATE INDEX search_idx ON topics USING gin(searchable_index_col);
+
+CREATE OR REPLACE FUNCTION private.update_searchable_index_col()
+    RETURNS TRIGGER
+    AS $$
```
Comment (erik-beus) [inline supabase/migrations/20241127203353_searchable_index_col.sql:17]: Consider installing the `pgFormatter` extension to format this code better 👍 https://marketplace.visualstudio.com/items?itemName=bradymholt.pgformatter

File: supabase/migrations/20241127203353_searchable_index_col.sql [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+DECLARE
+    message_content text;
+BEGIN
+    SELECT
+        m.content INTO message_content
+    FROM
+        public.messages m
+    WHERE
+        m.id = NEW.first_message_id;
+    NEW.searchable_index_col = to_tsvector('english', NEW.name || ' ' || message_content);
```
Comment (erik-beus) [inline supabase/migrations/20241127203353_searchable_index_col.sql:27]: Could `message_content` or `first_message_id` potentially be `null`?

**Review by erik-beus — CHANGES_REQUESTED** [review]

File: supabase/migrations/20241127203353_searchable_index_col.sql (hunk 4 above)
Comment (marija04) [reply supabase/migrations/20241127203353_searchable_index_col.sql:27]: Yes, the first_message_id can be null.

File: supabase/migrations/20241127203353_searchable_index_col.sql (hunk 1 above)
Comment (marija04) [reply supabase/migrations/20241127203353_searchable_index_col.sql:4]: Yes, I googled and seems that gin is preferred for tsvector type.

File: supabase/migrations/20241127203353_searchable_index_col.sql (hunk 3 above)
Comment (marija04) [reply supabase/migrations/20241127203353_searchable_index_col.sql:17]: I have it, this is how it looks after formatting it with `pgFormatter` :)

**Review by erik-beus — APPROVED** [review]

**Pull request merged**